src/main/java/cst438flights/controller/flightRestController.java

```diff
@@ -45,4 +47,39 @@
             return new ResponseEntity<List<Flight>>(flightInfo, HttpStatus.OK);
         }
     }
-}
\ No newline at end of file
+
+    @PostMapping("/api/flight/reservation")
+    public ResponseEntity<Reservation> createReservation(
+            @RequestParam("email") String email,
+            @RequestParam("seatClass") String seatClass,
+            @RequestParam("numPassengers") int numPassengers,
+            @RequestParam("prioBoarding") boolean prioBoarding,
+            @RequestParam("flightID") int flightID
+    ) {
+
+        Reservation reservation = flightService.requestReservation(email, seatClass, numPassengers, prioBoarding, ORIGIN, flightID);
+
+        if (reservation == null) {
+            return new ResponseEntity<Reservation>(HttpStatus.NOT_FOUND);
+        } else {
+            return new ResponseEntity<Reservation>(reservation, HttpStatus.OK);
+        }
+    }
+    
+    //getMapping option to book reservation
```

Comment:
Added a getMapping option. Gives planner team the use option and tests logic of PostMapping

Reply:
Tested to work in Postman